# HG changeset patch
# User Rik <rik@octave.org>
# Date 1386995148 28800
#      Fri Dec 13 20:25:48 2013 -0800
# Node ID 8ca7b1906a41506cfc1b1cc95fe595aa79549606
# Parent  c7b68a11074b80b76547ea88f60f410d1333440b
fzero.m: Correctly detct if OutputFcn is present (bug #40889).

* fzero.m: Use isempty to check whether OutputFcn is present.

diff --git a/scripts/optimization/fzero.m b/scripts/optimization/fzero.m
--- a/scripts/optimization/fzero.m
+++ b/scripts/optimization/fzero.m
@@ -297,17 +297,17 @@ function [x, fval, info, output] = fzero
       info = 1;
       break;
     else
       ## This should never happen.
       error ("fzero:bracket", "fzero: zero point is not bracketed");
     endif
 
     ## If there's an output function, use it now.
-    if (outfcn)
+    if (! isempty (outfcn))
       optv.funccount = nfev;
       optv.fval = fval;
       optv.iteration = niter;
       if (outfcn (x, optv, "iter"))
         info = -1;
         break;
       endif
     endif
